.rubocop.yml

```diff
@@ -312,5 +318,59 @@ Performance/DeleteSuffix:
 Performance/OpenStruct:
   Enabled: true
 
+Performance/InefficientHashSearch:
+  Enabled: true
+
+Performance/RedundantBlockCall:
+  Enabled: true
+
+Performance/AncestorsInclude:
+  Enabled: true
+
+Performance/BigDecimalWithNumericArgument:
+  Enabled: true
+
+Performance/BlockGivenWithExplicitBlock:
```

Comment:
I'm very doubtful about this one.

When you use `&block` the `Proc` object isn't allocated if you only forward `&block` to another method and don't manipulate in any other ways. I'm not 100% sure if `if block` cause the block to be allocated, I'd need to double check, but either way I don't think it's a good change.

Reply:
Ok, so `if block` and even `block.call` is properly optimized out. Still, I'm very much on the fence on that one.

Reply:
@byroot Wdyt on `block.call` vs `yield` changes on this PR - https://docs.rubocop.org/rubocop-performance/cops_performance.html#performanceredundantblockcall ? Same as `if block` vs `if block_given?`

Reply:
I don't mind it as much, because:

  - It increase clarity in case where the block is not called `block`
  - It's not that many changes, contrary to `block_given?`

Reply:
`if block` causes an allocation vs `block_given?`, so it's strange that rubocop-performance suggests it (I'm planning to fix this upstream soon). I see that's been removed so 👍

`block.call` and `yield` have extremely similar performance, is it really worth preferring one to the other?

Reply:
> if block causes an allocation vs block_given?

Are you sure? I thought the same and I tested it with a script, turns out it doesn't allocate. I think `if block` and `block.call` are both fine.

> block.call and yield have extremely similar performance, is it really worth preferring one to the other?

I somewhat like that change (not enough to die on that hill if other have different opinions) only because the block parameter isn't always called `block` (as you can see in this diff), so it's not always evident to me that `random_variable.call` is actually the passed block at first glance.

That said it's really a minor preference, so I don't mind.

Reply:
Hum, actually you are right, I'm not sure how I messed up my test the other day, but `if block` definitely allocates.

It's really surprising rubocop prefers it, but the more I see rubocop-performance cops, the more I'm doubting of their general quality.

Reply:
For the record:

```ruby
def alloc_count
  before = GC.stat(:total_allocated_objects)
  yield
  GC.stat(:total_allocated_objects) - before
end

def block_var_unused(&block)
  if block
    true
  end
end


# warm caches
alloc_count do
  10.times do
    block_var_unused {}
  end
end

count = alloc_count do
  10.times do
    block_var_unused {}
  end
end
p count
```

Reply:
Sent a patch upstream https://github.com/ruby/ruby/pull/6286